### ISSUE TITLE
Passing arguments with AutoRouter

### ISSUE THREAD

**Issue by kenyons** [FilledStacks/flutter-tutorials#102]
Can the AutoRouter still be used when arguments are passed?  I see how it is done manually in router of the navigation tutorial, but using the AutoRouter, the argument is not passed to the new view.  Can you update the navigation example to use the new stacked_services package?

**Comment by FilledStacks** [comment]
auto_route generates an argument class that you should use. It serialises it automatically. it's [ViewName]Arguments so if your view is called HomeView you should pass HomeViewArguments as your arguments when calling navigate. It will automatically serialise and pass that to your view

**Comment by kenyons** [comment]
Thanks, I had seen the arguments class generated for the HomeView, but had to run the build a couple of times for the other to show up for some reason.

One other question.  Any suggested way to architect a view that is basically a wrapper for a WebView in moving to the view/viewmodel architecture?  Right now I have just added a method on the viewmodel to be passed to the WebResourceErrorCallback on the WebView.  Doesn't seem to be a way to handle the HTTP request in the viewmodel and pass it to the WebView.

**Comment by FilledStacks** [comment]
@kenyons I don't think you can "pass" anything to the webview. You can listen for route changes on the webview and tell it where to navigate but other than that I don't think you can pass anything to the webview.
